hazelcast-hibernate/hazelcast-hibernate3/src/test/java/com/hazelcast/hibernate/RegionFactoryDefaultTest.java

```diff
@@ -46,6 +54,202 @@ protected Properties getCacheProperties() {
     }
 
     @Test
+    public void testNonStrictReadWriteEntity() {
+        Session session = null;
+        Transaction txn = null;
+
+        int entityCount = 10;
+        int childCount = 4;
+
+        insertDummyNonStrictRWEntities(entityCount, childCount);
+
+        sf.getCache().evictEntityRegions();
+        sf2.getCache().evictEntityRegions();
+
+        session = sf.openSession();
+        ArrayList<DummyEntityNonStrictRW> entities = new ArrayList<DummyEntityNonStrictRW>(entityCount);
+        for (int i=0; i<entityCount; i++) {
+            entities.add((DummyEntityNonStrictRW)session.get(DummyEntityNonStrictRW.class, (long)i));
+        }
+        assertEquals(entityCount*2, sf.getStatistics().getSecondLevelCacheMissCount());
+        assertEquals(0, sf.getStatistics().getSecondLevelCacheHitCount());
+
+        session.close();
+
+        session = sf.openSession();
+        txn = session.beginTransaction();
+        DummyEntityNonStrictRW updatedEntity = entities.get(0);
+        updatedEntity.setName("updated:" + 0);
+        session.update(updatedEntity);
+        DummyEntityNonStrictRW removedEntity = entities.get(1);
+        session.delete(removedEntity);
+        txn.commit();
+        session.close();
+
+        assertEquals(0, sf2.getStatistics().getSecondLevelCacheMissCount());
+        assertEquals(0, sf2.getStatistics().getSecondLevelCacheHitCount());
+
+        session = sf2.openSession();
+        entities = new ArrayList<DummyEntityNonStrictRW>(entityCount);
+        for (int i=0; i<entityCount; i++) {
+            DummyEntityNonStrictRW ent = (DummyEntityNonStrictRW)session.get(DummyEntityNonStrictRW.class, (long)i);
+            if(ent != null)
+                entities.add(ent);
+        }
+        assertEquals(entityCount-1, entities.size());
+        assertEquals(2 + childCount, sf2.getStatistics().getSecondLevelCacheMissCount());
+        assertEquals((entityCount - 1) * (1 + childCount) + childCount, sf2.getStatistics().getSecondLevelCacheHitCount());
+    }
+
+    @Test
+    public void testReadOnlyEntity() {
+        Session session = null;
```

Comment:
Too long. And don't declare your variables at the beginning. Declare them when you need them.